File: prompting_debug_assistant/bug_snippets/bug3.java
// bug3.java
// Category: Runtime Exceptions

import java.util.ArrayList;

public class Bug3 {

    public static void main(String[] args) {

        ArrayList<String> names = new ArrayList<>();

        names.add("Alice");
        names.add("Bob");

        for (int i = 0; i <= names.size(); i++) {
            System.out.println(names.get(i).toUpperCase());
        }

        String value = null;
        System.out.println(value.length());
    }
}

File: prompting_debug_assistant/bug_snippets/bug3.py
public class Bug3 {
    public static int divideNumbers(int a, int b) {
        return a / b;
    }

    public static void main(String[] args) {
        int x = 10;
        int y = 0;

        System.out.println("Result: " + divideNumbers(x, y));
    }
}